src/components/portfolio/DetailsMainAnalytics.tsx

```diff
@@ -28,12 +30,39 @@ type DetailsMainAnalyticsProps = {
   portfolio: NonEmptyPortfolioDetails;
 };
 
+type VolatilityLineData = {
+  [value: number]: string;
+};
+
 const DetailsMainAnalytics: React.FC<DetailsMainAnalyticsProps> = ({
   portfolio,
 }) => {
+  const { palette } = useTheme();
   const classes = useStyles();
   const { t } = useTranslation();
 
+  const sortedStocks: VolatilityLineData = {};
+
+  if (portfolio.positions) {
+    Object.values(portfolio.positions)
+      // sort by total value (largest to smallest)
+      .sort((a, b) => b.qty * b.stock.price - a.qty * a.stock.price)
+      // take the top 5
+      .slice(0, 4)
+      .map((p) => p.stock)
+      .forEach((s) => {
+        const val = Math.round(s.volatility * 100) / 100;
+
+        if (!sortedStocks[val]) {
+          // if there is no entry with this score create one
+          sortedStocks[val] = s.name;
+        } else {
+          // otherwise add on to an existing score
+          sortedStocks[val] = sortedStocks[val].concat(`\n${s.name}`);
+        }
+      });
+  }
+
```

Comment:
Isn't this the same as in DetailsMainAnalyst.tsx (lines 50 to 70)? This could be moved to a function in a different file and used by both components.